lib/Deserializer/functions.php

```diff
@@ -79,11 +90,15 @@ function keyValue(Reader $reader, $namespace = null) {
                 $values[$clark] = $reader->parseCurrentElement()['value'];
             }
         } else {
-            $reader->read();
+            if (!$reader->read()) {
+                break;
+            }
         }
     } while ($reader->nodeType !== Reader::END_ELEMENT);
 
-    $reader->read();
+    if (!$reader->read()) {
```

Comment:
Why was the `if` statement added here? This duplicates also the return of this method.

Reply:
this is true - thx for having a closer look. 

I added this for debugging purpose to see if read fails at this point under any condition .....